pretyping/tacred.ml

```diff
@@ -845,31 +833,15 @@ and whd_simpl_stack allowed_reds env sigma =
   in
   redrec
 
-and reduce_fix allowed_reds env sigma fix stack =
+and reduce_fix allowed_reds env sigma f fix stack =
   match fix_recarg fix stack with
     | None -> NotReducible
     | Some (recargnum,recarg) ->
        let (recarg'hd,_ as recarg') =
          whd_construct_stack allowed_reds env sigma recarg in
         let stack' = List.assign stack recargnum (applist recarg') in
         (match EConstr.kind sigma recarg'hd with
-           | Construct _ -> Reduced (contract_fix sigma fix, stack')
-           | _ -> NotReducible)
-
-and reduce_fix_use_function allowed_reds env sigma f fix stack =
-  match fix_recarg fix stack with
-    | None -> NotReducible
-    | Some (recargnum,recarg) ->
-        let (recarg'hd,_ as recarg') =
-          if EConstr.isRel sigma recarg then
-            (* The recarg cannot be a local def, no worry about the right env *)
-            (recarg, [])
```

Comment:
Can this codepath be taken? If some internal code calls `simpl` on, say, `Nat.add x 0` with `x := 0` in the de Bruijn environment, this could make a difference, right?

But even if a difference, is this worth? I'm not sure (we would like `Nat.add to reduce). So, ok for this change.


 conversely, is i

Reply:
It's unclear whether it matters in practice, but the code seems wrong to me. There is no reason to handle bound rels in a particular way. That said, the toplevel simpl command is doing so many weird things that I don't even see a way to really trigger this code path in an observable way.